apps/src/applab/designElements/TextAlignmentPropertyRow.jsx

```diff
@@ -0,0 +1,35 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import applabMsg from '@cdo/applab/locale';
+import EnumPropertyRow from './EnumPropertyRow';
+
+export const TEXT_ALIGNMENT_LEFT = 'left';
+export const TEXT_ALIGNMENT_RIGHT = 'right';
+export const TEXT_ALIGNMENT_CENTER = 'center';
+export const TEXT_ALIGNMENT_JUSTIFY = 'justify';
+
+const TEXT_ALIGNMENTS = [
+  TEXT_ALIGNMENT_LEFT,
+  TEXT_ALIGNMENT_RIGHT,
+  TEXT_ALIGNMENT_CENTER,
+  TEXT_ALIGNMENT_JUSTIFY,
+];
+
+export default function TextAlignmentPropertyRow(props) {
+  return (
+    <EnumPropertyRow
+      desc={applabMsg.designElementProperty_textAlignment()}
+      initialValue={props.initialValue}
+      options={TEXT_ALIGNMENTS}
+      displayOptions={TEXT_ALIGNMENTS.map(textAlignment =>
+        applabMsg[`designElementProperty_textAlignment_${textAlignment}`]()
```

Comment:
Do we do this composition to create the i18n key in other places? I like that the keys are easily searchable and not forced to be in a specific format when we spell them all out.

Reply:
In this particular case, the naming contains the context of the translation line.
It's not just `left` or `right` etc, it's `Left alignment` or `Right alignment` etc which could potentially have different words/meanings in different languages.

Reply:
Yeah, I'm talking about `designElementProperty_textAlignment_${textAlignment}` rather than `designElementProperty_textAlignment_right`

Reply:
Unfortunately, we have no other way to make it dynamic:
- https://github.com/code-dot-org/code-dot-org/blob/staging/apps/src/p5lab/AnimationPicker/AnimationPickerBody.jsx#L176-L178
- https://github.com/code-dot-org/code-dot-org/blob/staging/apps/src/p5lab/poetry/poem.js#L13-L14

Reply:
Interesting coincidence! I actually attempted to search the key `designElementTheme_ketchupAndMustard` and was surprised to not find it other than in `en_us.json`. I then noted the refactor (related to my comment above) but agree with Erin's point.

Reply:
@fisher-alice @elf-code, do you propose to not make it dynamic?

Reply:
My preference is static then dynamic if that's straightforward. By separating the full values of the keys, it's possible to have missing keys very easily. I think the two use-cases for dynamic keys make sense here. This one has the strings very closeby code-wise to deduce the strings for humans. The other one has the keys much farther away so is less ideal.